wagtail/admin/views/home.py

```diff
@@ -255,44 +255,27 @@ def get_context_data(self, parent_context):
 
         # Last n edited pages
         edit_count = getattr(settings, "WAGTAILADMIN_RECENT_EDITS_LIMIT", 5)
-        if connection.vendor == "mysql":
-            # MySQL can't handle the subselect created by the ORM version -
-            # it fails with "This version of MySQL doesn't yet support 'LIMIT & IN/ALL/ANY/SOME subquery'"
-            last_edits = Revision.objects.raw(
-                """
-                SELECT wr.* FROM
-                    wagtailcore_revision wr JOIN (
-                        SELECT max(created_at) AS max_created_at, object_id FROM
-                            wagtailcore_revision WHERE user_id = %s AND base_content_type_id = %s GROUP BY object_id ORDER BY max_created_at DESC LIMIT %s
-                    ) AS max_rev ON max_rev.max_created_at = wr.created_at ORDER BY wr.created_at DESC
-                 """,
-                [
-                    User._meta.pk.get_db_prep_value(request.user.pk, connection),
-                    get_default_page_content_type().id,
-                    edit_count,
-                ],
-            )
-        else:
-            last_edits_dates = (
-                Revision.page_revisions.filter(user=request.user)
-                .values("object_id")
-                .annotate(latest_date=Max("created_at"))
-                .order_by("-latest_date")
-                .values("latest_date")[:edit_count]
-            )
-            last_edits = Revision.page_revisions.filter(
-                created_at__in=last_edits_dates
-            ).order_by("-created_at")
-
-        # The revision's object_id is a string, so cast it to int first.
-        page_keys = [int(pr.object_id) for pr in last_edits]
-        pages = Page.objects.specific().in_bulk(page_keys)
-        context["last_edits"] = []
-        for revision in last_edits:
-            page = pages.get(int(revision.object_id))
+
+        # Query the audit log to get a resultset of (page ID, latest edit timestamp)
+        last_edits_dates = (
+            PageLogEntry.objects.filter(user=request.user, action="wagtail.edit")
+            .values("page_id")
+            .annotate(latest_date=Max("timestamp"))
+            .order_by("-latest_date")[:edit_count]
+        )
+        # Retrieve the page objects for those IDs
+        pages_mapping = Page.objects.specific().in_bulk(
+            [log["page_id"] for log in last_edits_dates]
+        )
+        # Compile a list of (latest edit timestamp, page object) tuples
+        last_edits = []
+        for log in last_edits_dates:
+            page = pages_mapping.get(log["page_id"])
             if page:
-                context["last_edits"].append([revision, page])
+                page.user_last_edited_at = log["latest_date"]
```

Comment:
This attribute doesn't seem to be used anywhere. https://github.com/wagtail/wagtail/issues/11323#issuecomment-1889520913 suggests that this would be used to sort the list, but it looks like the list is already sorted?

```suggestion
```